provisioner/terraform/serve.go

```diff
@@ -100,20 +99,14 @@ func Serve(ctx context.Context, options *ServeOptions) error {
 }
 
 type server struct {
-	// initMu protects against executors running `terraform init`
-	// concurrently when cache path is set.
-	initMu sync.Mutex
-
-	binaryPath string
-	cachePath  string
-	logger     slog.Logger
-
+	binaryPath  string
+	cachePath   string
+	logger      slog.Logger
 	exitTimeout time.Duration
 }
 
 func (s *server) executor(workdir string) executor {
 	return executor{
-		initMu:     &s.initMu,
```

Comment:
@deansheather how does this change help? We were already sharing the server mutex with the executors.

If there were multiple servers sharing a cache path I’d understand the issue. I had always imagined we’d use subfolders for each server/provisionerd instance.

Reply:
All provisioners in the same process share the same cache directory, and each had it's own mutex.

Reply:
Ah I misunderstood your PR description I think. This change is fine for now but I do think we should just have IDs for each provisioner and create subpaths in the cache dir in the future.

Reply:
Ah right, I think one reason I implemented it like this was for better test concurrency. Right now I believe most tests using the provisions have their own cache path. So this global mutex might slow those down due to not betting able to run concurrently or reuse the cache.